pkg/agent/config/traffic_encap_mode.go

```diff
@@ -30,10 +30,10 @@ const (
 
 var (
 	modeStrs = [...]string{
-		"Encap",
-		"NoEncap",
-		"Hybrid",
-		"NetworkPolicyOnly",
+		"encap",
+		"noEncap",
+		"hybrid",
+		"networkPolicyOnly",
```

Comment:
is this an intended change?
if it's needed for something else, you may want to merge that change separately into the main branch

Reply:
Yes, it is intended. The issue is revealed because in this function branch, `config.TrafficEncapModeNoEncap.String()` is used to directly compare values in `antrea.yml`.
https://github.com/antrea-io/antrea/blob/d899cd8a6bf430e98466642a61b8409eef27c5a2/cmd/antrea-agent/options.go#L293

I think the reason why it is ok on main is that for most of times,`GetTrafficEncapModeFromStr` is usde and `strings.EqualFold` in it is case-insensitive.
https://github.com/antrea-io/antrea/blob/d899cd8a6bf430e98466642a61b8409eef27c5a2/pkg/agent/config/traffic_encap_mode.go#L44

But I still think it should be changed to keep pace with values in `antrea.yml`.